### ISSUE TITLE
Debugger can't link against shared libs even if path/to/so is indicated.

### ISSUE THREAD

**Issue by Posto578** [microsoft/vscode-cpptools#1760]
I try to debug my application with _launch_ mode but it fails on linking shared libs. I indicated path to seek required lib. 

![image](https://user-images.githubusercontent.com/35492119/38144928-362c075e-3447-11e8-9691-f5e761c9c9e5.png)

OS: macOS High Sierra 10.13.3 
Vscode: Version 1.21.1 (1.21.1)
launch.json: https://pastebin.com/5ZAuXTKh
c_cpp_properties.json: https://pastebin.com/Mt9Vdsft
settings.json: https://pastebin.com/VwL4C9Vw

**Comment by sean-mcmanus** [comment]
Does your app run without the debugger? It sounds like it's a problem with your app or OS not having libboost installed.

**Comment by Posto578** [comment]
@sean-mcmanus Yes my app runs well without debugger. I put shared libs location to _.bash_profile_ `export DYLD_LIBRARY_PATH=/Users/miszo97/Documents/Programming/boost_1_66_0/stage/lib`
 and it worked but not in debugger case.

**Comment by Posto578** [comment]
@sean-mcmanus What is worth to mention:
I can run debugger in _attach mode_ I run my app and _attach mode_ works fine. All required libs are already installed on my mac.
The result of _otool -L awc.exe_  command.
```
awc.exe:
	libboost_system.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

**Comment by sean-mcmanus** [comment]
@pieandcakes and @WardenGnaw are the debugger guys -- they might know more.

**Comment by pieandcakes** [comment]
@Posto578 can you enable `"logging": { "engineLogging": true}` and send me that output? I want to see what lldb is returning when we tell it to use the additional search path.
